### ISSUE TITLE
Incompatible with stylelint 15.10.0

### ISSUE THREAD

**Issue by silverwind** [elirasza/stylelint-stylistic#12]
https://github.com/stylelint/stylelint/pull/6999 has removed a file imported by this module. I think it's better for the future that this plugin only relies on clearly public API in stylelint and not do file-level imports.

```
Error: Cannot find module 'stylelint/lib/utils/isRangeContextMediaFeature'
Require stack:
- node_modules/stylelint-stylistic/dist/rules/media-feature-name-case/index.js
- node_modules/stylelint-stylistic/dist/main.js
- node_modules/stylelint/lib/augmentConfig.js
- node_modules/stylelint/lib/createStylelint.js
- node_modules/stylelint/lib/resolveConfig.js
- node_modules/stylelint/lib/printConfig.js
- node_modules/stylelint/lib/cli.js
- node_modules/stylelint/bin/stylelint.js
```

**Comment by Mouvedia** [comment]
related:
- stylelint/stylelint#7003
- https://github.com/stylelint/stylelint/pull/7021#issuecomment-1615200305

**Comment by ybiquitous** [comment]
@silverwind Thanks for following up on the new Stylelint version!

At this point, I suggest the following:

- Use the `stylelint.utils.*` API as possible. That API is public, and the Stylelint team ensures compatibility.
- Copy utility files used to this repository. They're used internally, the Stylelint team doesn't ensure compatibility.
  - Of course, we may publish some of them from now on, but I cannot ensure it now.

**Comment by elirasza** [comment]
I originally wanted to avoid code duplicates, but your point is right. All utilities not found in the API will be moved in this project, others will be directed to the API. Thank you for your warning !

**Comment by karl-sjogren** [comment]
I would love to see this getting a quick fix so that I can close my dependabot PRs that want to upgrade stylelint so I've made a small PR that fixes the immediate problem in #13.

**Comment by elirasza** [comment]
Thanks to [karl-sjogren](https://github.com/karl-sjogren) in https://github.com/elirasza/stylelint-stylistic/pull/13, this has been fixed in [0.4.3](https://www.npmjs.com/package/stylelint-stylistic) :)
